drivers/eks/eks_driver.go

```diff
@@ -958,15 +957,30 @@ func (d *Driver) Update(ctx context.Context, info *types.ClusterInfo, options *t
 		return nil, err
 	}
 
+	var sendUpdate bool
 	if newState.KubernetesVersion != "" &&
 		newState.KubernetesVersion != state.KubernetesVersion {
+		sendUpdate = true
 		state.KubernetesVersion = newState.KubernetesVersion
 	}
 
-	if !reflect.DeepEqual(state, *oldstate) {
+	if newState.ClientID != "" && newState.ClientID != state.ClientID {
+		state.ClientID = newState.ClientID
+	}
+
+	if newState.ClientSecret != "" && newState.ClientSecret != state.ClientSecret {
+		state.ClientSecret = newState.ClientSecret
+	}
+
+	if sendUpdate {
 		if err := d.updateClusterAndWait(ctx, state); err != nil {
-			logrus.Errorf("error updating cluster: %v", err)
-			return info, err
+			equalVersionsErrorFmt := "Unsupported Kubernetes minor version update from %s to %s"
+			if !strings.Contains(err.Error(), fmt.Sprintf(
```

Comment:
This seems somewhat brittle, if aws changes their error string it will fail again. Not sure of better solution, can we ping the cluster to check its version ? Otherwise we should at least lower the string.

Reply:
we can try getting cluster data and checking

Reply:
Decided that we are going to solve this problem with future work, as requirements already include implementing a state sync.